apps/profile/app/[locale]/onboarding/page.tsx

```diff
@@ -61,21 +64,49 @@ export default async (props: { params: { locale: string } }) => {
 
   const userLogs = (await userLogResponse.json()) as SignInEventResponse[]
 
-  const mygovid = userLogs
-    ?.at(0)
-    ?.payload.interaction?.verificationRecords?.find(
-      (i) => i.connectorId === "mygovid",
+  let safeLevel: number | undefined
+
+  if (!userLogs.length) {
+    const userResponse = await fetch(
+      url.logto.user(config.logtoEndpoint, loggedInUser.id),
+      {
+        headers: {
+          "Content-Type": "application/json",
+          Authorization: `Bearer ${accessToken}`,
+        },
+      },
     )
 
-  if (!mygovid) {
-    return <NotAuthorized />
+    if (userResponse.status === 200) {
+      const user = (await userResponse.json()) as MyGovIDUserResponse
+      const level = Number(
+        user?.identities?.[myGovIDIdentityKey]?.details?.rawData
+          ?.DSPOnlineLevel,
+      )
+      safeLevel = Number.isNaN(level) ? undefined : level
+    } else {
+      logger.error(
+        {},
+        "failed to get logto user when onboarding a first time mygovid user",
+      )
+    }
+  } else {
+    const level = Number(
+      userLogs
+        ?.at(0)
+        ?.payload.interaction?.verificationRecords?.find(
+          (i) => i.connectorId === "mygovid",
+        )?.socialUserInfo.rawData.DSPOnlineLevel,
+    )
+
+    safeLevel = Number.isNaN(level) ? undefined : level
```

Comment:
could it return a string? 
```suggestion
    safeLevel = Number.isNaN(level) ? undefined : Number(level)
```